### ISSUE TITLE
twisted.names.server lacks documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6886]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)| @wallrj reported|
|-|-|
|Trac ID|trac#6886|
|Type|enhancement|
|Created|2013-12-30 13:30:16Z|
|Branch|https://github.com/twisted/twisted/tree/names-server-documentation-6886|

Add missing API docstrings to twisted.names.server

Broken out from [#6700](https://github.com/twisted/twisted/issues/6700)

<details><summary>Searchable metadata</summary>

```
trac-id__6886 6886
type__enhancement enhancement
reporter__rwall rwall
priority__normal normal
milestone__ 
branch__branches_names_server_documentation_6886 branches/names-server-documentation-6886
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1388410216000000 1388410216000000
changetime__1388746115000000 1388746115000000
version__None None
owner__rwall rwall

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_1"></a>|@wallrj commented|
|-|-|

(In [41031]) Branching to 'names-server-documentation-6886'

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj removed owner|
|-|-|

Ready for review in log:branches/names-server-documentation-6886
 * I created a new branch and merged in the docstring changes from [#6700](https://github.com/twisted/twisted/issues/6700)
 * Addressed most of exarkun's review comments (ticket:6700#comment:6)

Forced another build which looks clean:
 * https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/names-server-documentation-6886

Some more responses to the last code review below...

Replying to [ticket:6700#comment:6 exarkun]:
>   1. twisted/names/server.py
>     1. `DNSServerFactory.cache` can also be `None` if no caches are specified.

r41034

>     1. For `DNSServerFactory.clients`, when referencing an interface, I prefer
>     to always be explicit about "implementer" or "provider".  This is the
>     vocabulary `zope.interface` uses.  Typically the class is an implementer
>     and instances of the class are providers (but things can be defined in
>     other ways using other declaration apis from `zope.interface`).  Examples
>     of this ambiguity appear elsewhere in the diff as well.

r41035 (also added provider elsewhere)

>     1. I don't think the _@type_ for `DNSServerFactory.protocol` adds much
>     to the _@ivar_.  I even find it a bit confusing if interpreted
>     literally: the callable should not actually return `IProtocolFactory`.  It
>     should return an _IProtocolFactory provider_.  Also _L{callable}_ is
>     incorrect here.  The `callable` builtin is just some random other API not
>     related to the type of this instance attribute.

r41035 (I said "IProtocolFactory constructor" instead)

>     1. If you care about Python 3 compat, then the signature of
>     `DNSServerFactory.gotResolverResponse` is a problem (tuple unpacking in
>     argument lists was removed from the language).  Not new in this branch but
>     the function is being changed here.

Yeah, I think I'll leave that. I've also reverted my incorrect change to the
wrapping in that function too.

41036.

>     1. In `DNSServerFactory.gotResolverError`, ''@param failure: A C{failure}
>     ..._ - I'm not sure what useful information _C{failure}'' is conveying.
>     You might want to move the _L{...}_ from the _@type failure_ into the
>     _@param_ to replace _C{failure}_.  Or you could be more abstract - eg,
>     ''The reason for the failed resolution (as reported by
>     C{self.resolver.query}).''  There are other instances of this elsewhere in
>     the patch as well, eg _C{deferred}_ in `handleQuery`.

r41037 (Couldn't see any other examples of C{failure})

>   1. The wrapping applied throughout seems to be too narrow.

r41039. (Done. I had my editor set up wrong when I created the original branch)

Thanks again for the review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @wallrj|
|-|-|

Thanks.  Just one last comment - in `DNSServerFactory.__init__`'s docstring, _@type clients: L{list} of {IResolver} providers_ is missing an _L_ before _{IResolver}_.  Please address that and then merge.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj set status to `closed`|
|-|-|

(In [41100]) Merge names-server-documentation-6886

Author: rwall
Reviewer: exarkun
Fixes: [#6886](https://github.com/twisted/twisted/issues/6886)
Refs: [#6700](https://github.com/twisted/twisted/issues/6700)

Add API documentation to twisted.names.server
